File: Laba6/Laba6/Models/CustomData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba6.Models
{
    public class CustomData
    {
        public int Day;
        public int Month;
        public int Year;

        public CustomData()
        {
            this.Day = 1;
            this.Month = 1;
            this.Year = 1;
        }

        public CustomData(int day, int month, int year)
        {
            Day = day;
            Month = month;
            Year = year;
        }

        public void IncreaseYear()
        {
            Year++;
        }

        public void DecreaseDayBy2()
        {
            if (Day > 2)
            {
                Day -= 2;
            }
            else
            {
                if (Month > 1)
                {
                    Month--;
                    Day = 30;
                }
                else
                {
                    if (Year > 0)
                    {
                        Year--;
                        Month = 12;
                        Day = 30;
                    }
                }
            }
        }


        public override string ToString()
        {
            return $"Day: {Day}, Month: {Month}, Year: {Year}";
        }

        ~CustomData()
        {
            Console.WriteLine("Об'єкт зтертий з пам'яті");
        }
    }
}

File: Laba6-Python/TaskModels/customData.py
class CustomData:
    def __init__(self, day=1, month=1, year=1):
        self.Day = day
        self.Month = month
        self.Year = year

    def increase_year(self):
        self.Year += 1

    def decrease_day_by_2(self):
        if self.Day > 2:
            self.Day -= 2
        else:
            if self.Month > 1:
                self.Month -= 1
                self.Day = 30
            else:
                if self.Year > 0:
                    self.Year -= 1
                    self.Month = 12
                    self.Day = 30

    def __str__(self):
        return f"Day: {self.Day}, Month: {self.Month}, Year: {self.Year}"

    def __del__(self):
        print("Object deleted from memory")